src/ageml/modelling.py

```diff
@@ -367,105 +367,127 @@ def fit_age(self, X, y):
         if self.pipeline is None and self.model_type != "hyperopt":
             raise TypeError("Must set a valid pipeline before running fit.")
 
-        # Variables of interes
-        pred_age = np.zeros(y.shape)
-        corrected_age = np.zeros(y.shape)
-        metrics_train = []
-        metrics_test = []
-
-        # Optimize hyperparameters if required
-        if self.hyperparameter_grid != {}:
-            print("Running Hyperparameter optimization...")
-            opt_pipeline = model_selection.GridSearchCV(self.pipeline,
-                                                        self.hyperparameter_grid,
-                                                        cv=self.CV_split,
-                                                        scoring="neg_mean_absolute_error")
-            opt_pipeline.fit(X, y)
-            print(f"Hyperoptimization best parameters: {opt_pipeline.best_params_}")
-            # Set best parameters in pipeline
-            self.pipeline.set_params(**opt_pipeline.best_params_)
+        if self.model_type != "hyperopt":
+            # Optimize hyperparameters if required
+            if self.hyperparameter_tuning > 0:
```

Comment:
hyperparameter tunnning should be at least 2 points? will give an error if you have C=1,2 and only have one point. Will it choose C=1 or C=2?

Reply:
True